sklearn/model_selection/_split.py

```diff
@@ -543,11 +543,11 @@ class StratifiedKFold(_BaseKFold):
     >>> from sklearn.model_selection import StratifiedKFold
     >>> X = np.array([[1, 2], [3, 4], [1, 2], [3, 4]])
     >>> y = np.array([0, 0, 1, 1])
-    >>> skf = StratifiedKFold(n_splits=2)
+    >>> skf = StratifiedKFold(n_splits=2, random_state=0)
```

Comment:
Okay so we want `repr` to preserve the user set random state or the default of `None` if `shuffle` is `False`, correct? In other cases set an explicit `random_state` if `None` and display the explicitly set `random_state` in `repr`... For the tests use ellipsis to suppress the test error. Sounds good?